src/lib/markbind/src/parser.js

```diff
@@ -401,29 +235,15 @@ class Parser {
         decodeEntities: true,
       });
 
-      const parentSitePath = urlUtils.getParentSiteAbsolutePath(file, config.rootPath, config.baseUrlMap);
-      const userDefinedVariables = config.userDefinedVariablesMap[parentSitePath];
-      const pageVariables = Parser.extractPageVariables(file, content, userDefinedVariables, {});
-      let fileContent = njUtil.renderRaw(content, {
-        ...pageVariables,
-        ...userDefinedVariables,
-        ...additionalVariables,
-      });
-      this._extractInnerVariables(fileContent, context, config);
-      const innerVariables = this.getImportedVariableMap(context.cwf);
-      fileContent = njUtil.renderRaw(fileContent, {
-        ...userDefinedVariables,
-        ...additionalVariables,
-        ...innerVariables,
-      });
+      const renderedContent = this.variablePreprocessor.renderPage(file, content, additionalVariables);
```

Comment:
huge boilerplate reduction, due to the process overhaul and method extraction.
The left shows the lengthy 2 step process as mentioned in the overview in the op.

This is also the first of 2 methods belonging to function 4 I changed. (the second being `renderIncludeFile` (annotated below shortly))